program/rust/src/rust_oracle.rs

```diff
@@ -227,6 +220,118 @@ pub fn add_mapping(
     Ok(SUCCESS)
 }
 
+/// a publisher updates a price
+/// accounts[0] publisher account                                   [signer writable]
+/// accounts[1] price account to update                             [writable]
+/// accounts[2] sysvar clock                                        []
+pub fn upd_price(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    instruction_data: &[u8],
+) -> OracleResult {
+    let cmd_args = load::<cmd_upd_price_t>(instruction_data)?;
+
+    let [funding_account, price_account, clock_account] = match accounts {
+        [x, y, z] => Ok([x, y, z]),
+        [x, y, _, z] => Ok([x, y, z]),
+        _ => Err(ProgramError::InvalidArgument),
+    }?;
+
+    check_valid_funding_account(funding_account)?;
+    check_valid_writable_account(program_id, price_account, size_of::<pc_price_t>())?;
+    // Check clock
+    let clock = Clock::from_account_info(clock_account)?;
+
+    let mut publisher_index: usize = 0;
+    let latest_aggregate_price: pc_price_info_t;
+    let latest_publisher_price: pc_price_info_t;
```

Comment:
I don't think this needs to be declared outside the block below

Reply:
`publisher_index` and `latest_aggregate_price ` are you used afterwards. `latest_publisher_price ` can be declared inside.